### ISSUE TITLE
Add an option to allow the purchase with already registered emails.

### ISSUE THREAD

**Issue by jesus997** [woocommerce/woocommerce#26942]
**Is your feature request related to a problem? Please describe.**
I'm currently building a store in which I won't need my clients to log in but I do need them to register. It's a bit confusing, I know, but the goal is that you can enable login later and customers who have already purchased don't have to re-register and lose their purchase history. I currently have this configuration in WooCommerce > Settings > Accounts & Privacy:

***Guest checkout***
- [x] Allow customers to place orders without an account 
- [ ] Allow customers to log into an existing account during checkout 

***Account creation***
- [x] Allow customers to create an account during checkout 
- [ ] Allow customers to create an account on the "My account" page 
- [x] When creating an account, automatically generate an account username for the customer based on their name, surname or email 
- [x] When creating an account, automatically generate an account password 

The problem with this is that when an existing customer tries to make another purchase they receive this error message: 

> An account is already registered with your email address. Please log in.

I know it's logical for it to work like it does now, but for some custom projects an option to disable this behavior may come in handy.

**Describe the solution you'd like**
I found that the function `wc_create_new_customer` create a new user when an order is created so I think we could modify there to add a new conditional, something like this:

```php
/**
 * Create a new customer.
 *
 * @param  string $email    Customer email.
 * @param  string $username Customer username.
 * @param  string $password Customer password.
 * @param  array  $args     List of arguments to pass to `wp_insert_user()`.
 * @return int|WP_Error Returns WP_Error on failure, Int (user ID) on success.
 */
function wc_create_new_customer( $email, $username = '', $password = '', $args = array() ) {
    if ( empty( $email ) || ! is_email( $email ) ) {
        return new WP_Error( 'registration-error-invalid-email', __( 'Please provide a valid email address.', 'woocommerce' ) );
    }

    $registered_customer_id = email_exists( $email );

    if( 'yes' ===  get_option( 'woocommerce_allow_purchases_with_existing_email', 'no' ) && $registered_customer_id) {
        return $registered_customer_id;
    }

    if ( $registered_customer_id ) {
        return new WP_Error( 'registration-error-email-exists', apply_filters( 'woocommerce_registration_error_email_exists', __( 'An account is already registered with your email address. <a href="#" class="showlogin">Please log in.</a>', 'woocommerce' ), $email ) );
    }

    // ... default logic

    return $customer_id;
}
```

Basically the logic behind this is that if a user already exists with the given email we return the id of that user instead of creating a new one. I am considering adding a new option to enable or disable this functionality (`woocommerce_allow_purchases_with_existing_email`). By default it would be disabled for compatibility reasons.

**Describe alternatives you've considered**
The simplest alternative is to enable login at checkout.

**Additional context**
This function uses `function_exists ('wc_create_new_customer')` but it cannot be overwritten from a theme, I already tried that.

**Comment by rodrigokr** [comment]
Hello, did you have a solution to enable purchase without registration?

**Comment by jesus997** [comment]
> Hello, did you have a solution to enable purchase without registration?

Probably yes, but it's been a long time since I worked on that project, so I don't remember it anymore. I'm sorry.

**Comment by rodrigokr** [comment]
I made a new code and it worked. Here are the steps and code:
The code gets the order email.
1- It tries to find an existing user with that email using the function get_user_by('email', $email).
2 - If an existing user is found ($user is not null), the code simply gets the ID of that existing user.
3 - If no existing user is found ($user is null), the code creates a new user with the order email and a randomly generated password. It then gets the new user ID.
4 -Finally, the code links the order to the user (existing or newly created) by updating the order meta with the user ID.

** Some functions are in Portuguese, I think it's better to change that.**

add_action('woocommerce_new_order', 'vincular_pedido_a_conta',  1, 1);
add_action('woocommerce_checkout_update_order_meta', 'vincular_pedido_a_conta');

function vincular_pedido_a_conta($order_id) {
    $order = new WC_Order($order_id);
    $email = $order->get_billing_email();
    $user  = get_user_by('email', $email);

    if (!$user) {
        $password = wp_generate_password();
        $userdata = array(
            'user_login' => $email,
            'user_email' => $email,
            'user_pass'  => $password,
            'role'       => 'customer',
        );

        $user_id = wp_insert_user($userdata);

        if (!is_wp_error($user_id)) {
            wp_update_user(array('ID' => $user_id, 'first_name' => $order->get_billing_first_name(), 'last_name' => $order->get_billing_last_name()));
        }
    } else {
        $user_id = $user->ID;
    }

    update_post_meta($order_id, '_customer_user', $user_id);
    $order->set_customer_id($user_id);
    $order->save();

    // Fazer o login do usuário automaticamente
    wp_set_current_user($user_id);
    wp_set_auth_cookie($user_id);
}

**Comment by jamesckemp** [comment]
This issue has been automatically closed as part of the backlog cleanup announced in [Refining how we manage the WooCommerce backlog](https://developer.woocommerce.com/2026/05/29/backlog-cleanup/). It met the criteria outlined there. If this still affects a current version of WooCommerce, please follow the steps in that post. A fresh report with current reproduction steps is the fastest path to triage. If this is a feature request, please add it to the [WooCommerce feature request board](https://woocommerce.com/feature-requests/woocommerce/) so it can be considered alongside other requests.
